src/utils/events/index.ts

```diff
@@ -3,63 +3,173 @@ import {
   AbiEnums,
   AbiEvents,
   AbiStructs,
-  Cairo1Event,
+  Cairo1Events,
   EventAbi,
   LegacyEvent,
   ParsedEvent,
   ParsedEvents,
   RPC,
+  type Cairo1Event,
+  type Cairo1EventVariant,
 } from '../../types';
+import assert from '../assert';
+import { isCairo1Abi } from '../calldata/cairo';
 import responseParser from '../calldata/responseParser';
 import { starkCurve } from '../ec';
 import { addHexPrefix, utf8ToArray } from '../encode';
 
 /**
  * Retrieves the events from the given ABI.
  *
+ * Is able to handle Cairo 1 events nested in Cairo components.
  * @param {Abi} abi - The ABI to extract events from.
  * @return {AbiEvents} - An object containing the extracted events.
+ * @example
+ * ```typescript
+ * const result = events.getAbiEvents(abi);
+ * // result = {
+ * //   '0x22ea134d4126804c60797e633195f8c9aa5fd6d1567e299f4961d0e96f373ee': 
+ * //    { '0x34e55c1cd55f1338241b50d352f0e91c7e4ffad0e4271d64eb347589ebdfd16': {
+ * //     kind: 'struct', type: 'event',
+ * //     name: 'ka::ExComponent::ex_logic_component::Mint',
+      
+ * //     members: [{
+ * //      name: 'spender',
+ * //      type: 'core::starknet::contract_address::ContractAddress',
+ * //      kind: 'key'},
+ * //      { name: 'value', type: 'core::integer::u256', kind: 'data' }]},
+ * // ...
+ * ```
  */
 export function getAbiEvents(abi: Abi): AbiEvents {
-  return abi
-    .filter((abiEntry) => abiEntry.type === 'event' && (abiEntry.size || abiEntry.kind !== 'enum'))
-    .reduce((acc, abiEntry) => {
-      const entryName = abiEntry.name.slice(abiEntry.name.lastIndexOf(':') + 1);
-      const abiEntryMod = { ...abiEntry };
-      abiEntryMod.name = entryName;
-      return {
-        ...acc,
-        [addHexPrefix(starkCurve.keccak(utf8ToArray(entryName)).toString(16))]: abiEntryMod,
-      };
-    }, {});
+  let finalResult: AbiEvents;
+  if (!isCairo1Abi(abi)) {
+    // Cairo 0 abi
+    finalResult = abi
+      .filter((abiEntry) => abiEntry.type === 'event')
+      .reduce((acc, abiEntry) => {
+        const entryName = abiEntry.name;
+        const abiEntryMod = { ...abiEntry };
+        abiEntryMod.name = entryName;
+        return {
+          ...acc,
+          [addHexPrefix(starkCurve.keccak(utf8ToArray(entryName)).toString(16))]: abiEntryMod,
+        };
+      }, {});
+  } else {
+    // Cairo 1 abi
+    const abiEventsStructs = abi.filter((obj) => obj.type === 'event' && obj.kind === 'struct');
```

Comment:
` obj.type === 'event'`

could be extracted as a helper, similar to isString or isObject logic ?

Reply:
OK. Where to locate this function?

Reply:
I thought of maybe the same file, but this is nitpicking, since we also have `obj.kind === 'struct'` etc... so ok, let's leave it like this unless in the future there is a need to extract to other methods :)